wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -0,0 +1,2592 @@
+					}
+				}
+			} while ( $locked );
+
+			if ( $status > 0 ) {
+				$message                = sprintf(
+					'<p>%s</p><p>%s</p><p>%s</p>',
+					'Database initialization error!',
+					"Code: $status",
+					"Error Message: $err_message"
+				);
+				$this->is_error         = true;
+				$this->error_messages[] = $message;
+				return;
+			}
+		}
+
+		WP_SQLite_PDO_User_Defined_Functions::register_for( $pdo );
+
+		// MySQL data comes across stringified by default.
+		$pdo->setAttribute( PDO::ATTR_STRINGIFY_FETCHES, true );
+
+		$this->pdo = $pdo;
+
+		$this->information_schema_builder = new WP_SQLite_Information_Schema_Builder(
+			$this->db_name,
+			array( $this, 'execute_sqlite_query' )
+		);
+		$this->information_schema_builder->ensure_information_schema_tables();
+
+		// Load MySQL grammar.
+		if ( null === self::$grammar ) {
+			self::$grammar = new WP_Parser_Grammar( require self::GRAMMAR_PATH );
+		}
+
+		// Fixes a warning in the site-health screen.
+		// @TODO: It is not clear how this fixes a warning or if it serves a
+		//        different purpose. We should also make this use PDO and
+		//        consider providing that information lazily via a getter.
+		$this->client_info = SQLite3::version()['versionString'];
+
+		$this->pdo->query( 'PRAGMA foreign_keys = ON' );
+		$this->pdo->query( 'PRAGMA encoding="UTF-8";' );
+
+		$valid_journal_modes = array( 'DELETE', 'TRUNCATE', 'PERSIST', 'MEMORY', 'WAL', 'OFF' );
+		if ( defined( 'SQLITE_JOURNAL_MODE' ) && in_array( SQLITE_JOURNAL_MODE, $valid_journal_modes, true ) ) {
+			$this->pdo->query( 'PRAGMA journal_mode = ' . SQLITE_JOURNAL_MODE );
+		}
+	}
+
+	/**
+	 * Get the PDO object.
+	 *
+	 * @return PDO
+	 */
+	public function get_pdo() {
+		return $this->pdo;
+	}
+
+	/**
+	 * Method to return inserted row id.
+	 */
+	public function get_insert_id() {
+		return $this->last_insert_id;
+	}
+
+	/**
+	 * Method to return the number of rows affected.
+	 */
+	public function get_affected_rows() {
+		return $this->affected_rows;
+	}
+
+	/**
+	 * Translate and execute a MySQL query in SQLite.
+	 *
+	 * A single MySQL query can be translated into zero or more SQLite queries.
+	 *
+	 * @param string $query              Full SQL statement string.
+	 * @param int    $fetch_mode         PDO fetch mode. Default is PDO::FETCH_OBJ.
+	 * @param array  ...$fetch_mode_args Additional fetch mode arguments.
+	 *
+	 * @return mixed        Return value, depending on the query type.
+	 * @throws Exception    If the query could not run.
+	 * @throws PDOException If the translated query could not run.
+	 */
+	public function query( string $query, $fetch_mode = PDO::FETCH_OBJ, ...$fetch_mode_args ) {
+		$this->flush();
+		if ( function_exists( 'apply_filters' ) ) {
+			/**
+			 * Filters queries before they are translated and run.
+			 *
+			 * Return a non-null value to cause query() to return early with that result.
+			 * Use this filter to intercept queries that don't work correctly in SQLite.
```

Comment:
Why intercept them? Is this for error reporting?

Reply:
I'm removing these hooks from the new driver for now. We can add them back in the future if needed.